src/CMakeLists.txt

```diff
@@ -5,14 +5,31 @@ else()
   add_library(${PROJECT_NAME} SHARED ${FCL_HEADERS} ${FCL_SOURCE_CODE})
 endif()
 
-target_link_libraries(${PROJECT_NAME} ${CCD_LIBRARIES} ${OCTOMAP_LIBRARIES} ${Boost_LIBRARIES})
-set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${FCL_VERSION} SOVERSION ${FCL_ABI_VERSION})
+set_target_properties(${PROJECT_NAME} PROPERTIES
+  VERSION ${FCL_VERSION}
+  SOVERSION ${FCL_ABI_VERSION})
+
+target_link_libraries(${PROJECT_NAME}
+  PUBLIC ${OCTOMAP_LIBRARIES}
+  PRIVATE ${CCD_LIBRARIES}
+  PRIVATE ${Boost_LIBRARIES})
+
+target_include_directories(${PROJECT_NAME} INTERFACE
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+
+export(TARGETS ${PROJECT_NAME}
+  FILE "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
 
 install(TARGETS ${PROJECT_NAME}
+  EXPORT ${PROJECT_NAME}Config
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
+install(EXPORT ${PROJECT_NAME}Config
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
```

Comment:
With this installation location, `fclConfig.cmake` will not be found automatically on Windows even if the system `CMAKE_PREFIX_PATH` enviromental variable matches the `CMAKE_INSTALL_PREFIX` of `fcl`. [1]
Modifying the code as in [2] :
~~~
if(WIN32 AND NOT CYGWIN)
  set(_EXPORT_INSTALL_DESTINATION CMake)
else()
  set(_EXPORT_INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
endif()

install(EXPORT ${PROJECT_NAME}Config
        DESTINATION ${_EXPORT_INSTALL_DESTINATION})
~~~ 
will ensure that the library will be easily found also in Windows.

[1] : https://cmake.org/cmake/help/v3.0/command/find_package.html
[2] : https://github.com/robotology/ycm/blob/master/modules/InstallBasicPackageFiles.cmake#L176